src/jetstream.rs

```diff
@@ -603,7 +608,125 @@ impl JetStream {
         Self { nc, options }
     }
 
-    /// Adds new stream to `JetStream`.
+    /// Publishes a message to `JetStream`
+    pub fn publish(&self, subject: &str, data: impl AsRef<[u8]>) -> io::Result<PublishAck> {
+        self.publish_with_options_or_headers(subject, None, None, data)
+    }
+
+    /// Publishes a message to `JetStream` with the given options.
+    pub fn publish_with_options(
+        &self,
+        subject: &str,
+        data: impl AsRef<[u8]>,
+        options: &PublishOptions,
+    ) -> io::Result<PublishAck> {
+        self.publish_with_options_or_headers(subject, Some(options), None, data)
+    }
+
+    /// Publishes a `Message` to `JetStream`.
+    pub fn publish_message(&self, message: &Message) -> io::Result<PublishAck> {
+        self.publish_with_options_or_headers(&message.subject, None, None, &message.data)
```

Comment:
Message may contain Headers, yet we ignore them here.

Reply:
Yikes, good catch.